### DISCUSSION TITLE
How to work with development version

### DISCUSSION THREAD

**Discussion by treagod** [martenframework/marten#135] [Q&A] [answered]
Hi!

I wanted to tackle #112. But to confirm that everything is working as expected, I need to create a demo project with the latest dependencies (that is pre-0.4, not 0.3).

How do you do this? I wanted to update the `shards.yml` to use the main branches of `marten` and `marten-auth`, but this throws a dependency error. Another alternative I have in mind is to symlink both projects in the `lib/` folder.

I would be happy to hear your solution so I can start working on the problem. This would also be a great addition to the developer documentation.

**Comment by ellmetha** [comment, marked as answer]
Hey! 👋 

It really depends on what you are developing/testing, but usually there are two main ways to do it:

1. You can symlink your local Marten repository as you already mentioned.
2. You can keep the latest stable version of Marten referenced in the `shard.yml` file but at the same time you can define a `shard.override.yml` file in which you point to your development version of Marten:

```yaml
# file: shard.override.yml
dependencies:
  marten:
    path: /path/to/marten
```

(This capability is also documented in the [official shards documentation](https://crystal-lang.org/reference/1.10/man/shards/index.html#fixing-dependency-version-conflicts))

If you use (2) and then run `shards install`, your project will use the overridden shard instead of the version defined in the `shard.yml` file.

I also agree that the second solution is something we should probably add to the developer documentation. 🙂
